### PR TITLE
WIP on retrieving the mbox files from the fedora archives.

### PR DESCRIPTION
**Pull request by cdolfi**
This is a WIP on getting the mbox from fedora archives. I am looking to see if there is a way to do this without having to save locally. Also, I have ran into a small issue that at times that sometimes the gzip sometimes does not always get the entire archive but that also happens at times when you do it manually. The last two tries worked successfully.

### CODE REVIEW

File: notebooks/retrieve_mbox.ipynb [hunk 1]
```diff
@@ -0,0 +1,165 @@
+{
```
Comment (oindrillac) [inline notebooks/retrieve_mbox.ipynb:1]: <p>@cdolfi can you create a Pipfile in your repository with all the required dependencies for this notebook and the parsing_mbox notebook?</p><p>Here's how you can use Pipenv within Jupyterhub <a href="https://docs.google.com/document/d/1BUEzAeymOr1NyWQT4_vY22dFlMinjcbeV6iFBZhBTYY/edit" target="_blank">https://docs.google.com/document/d/1BUEzAeymOr1NyWQT4_vY22dFlMinjcbeV6iFBZhBTYY/edit</a></p><p>let me know if you have any questions :)</p><p> </p><p>Also can you remove the import statements which are not being used in the notebook eg requests? maybe @sesheta bot can help with notebook linting</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/fedora-mailing-list-analysis/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (oindrillac) [inline notebooks/retrieve_mbox.ipynb:1]: <p>I'm not sure we can directly read into the bucket. Downloading locally and saving into the bucket, would be sufficient IMO</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/fedora-mailing-list-analysis/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Comment by oindrillac** [comment]
@cdolfi I think your changes look good :tada: I think this notebook looks ready to add after making the minor suggested changes(please see comments).

File: notebooks/retrieve_mbox.ipynb (hunk 1 above)
Comment (MichaelClifford) [reply notebooks/retrieve_mbox.ipynb:1]: <p>+1 to what @oindrillac said. I don't really see the need for writing directly to s3. But I found this stack-exchnage article that might be helpful if you  want to do this. https://stackoverflow.com/questions/40336918/how-to-write-a-file-or-data-to-an-s3-object-using-boto3/40336919 </p><p> </p><p>Something like:</p><pre class="ql-syntax" spellcheck="false">import boto3

s3 = boto3.resource(
    's3',
    region_name='us-east-1',
    aws_access_key_id=KEY_ID,
    aws_secret_access_key=ACCESS_KEY
)
content="String content to write to a new S3 file"
s3.Object('my-bucket-name', 'newfile.txt').put(Body=content)
</pre><p> </p>

**Comment by oindrillac** [comment]
@cdolfi thanks for adding the Pipfile. For notebook linting you can use [flake8-nb](https://pypi.org/project/flake8-nb/). On a quick glance I couldnt spot many errors other than the unused import statements, but running it through a lint check might help.
Once that is done, you can remove the WIP tag from the PR, and it should be ready for review/merge.

**Pull request merged**